src/Features/CSharp/Portable/ChangeSignature/CSharpChangeSignatureService.cs

```diff
@@ -450,6 +454,59 @@ private SyntaxNode GetNodeContainingTargetNode(SyntaxNode matchingNode)
             return null;
         }
 
+        private T UpdateArgumentList<T>(
+            ISymbol declarationSymbol,
+            SignatureChange signaturePermutation,
+            T argumentList,
+            bool isReducedExtensionMethod,
+            bool isParamsArrayExpanded) where T : BaseArgumentListSyntax
+        {
+            var newArguments = PermuteArgumentList(
+                declarationSymbol,
+                argumentList.Arguments,
+                signaturePermutation.WithoutAddedParameters(),
+                isReducedExtensionMethod);
+
+            newArguments = AddNewArgumentsToList(
+                declarationSymbol,
+                newArguments,
+                argumentList.Arguments,
+                signaturePermutation,
+                isReducedExtensionMethod,
+                isParamsArrayExpanded,
+                generateAttributeArguments: false);
+
+            return (T)argumentList
```

Comment:
might be good to just have an example in comments of what the two steps above this do.

Reply:
Done